### PR TITLE
fix(2056): Add webhook after create child pipeline

### PR DESCRIPTION
**Pull request by kumada626**
## Context

We should add webhook after create child pipeline.

## Objective

Add webhook and sync the child pipeline after create it.

## References

screwdriver-cd/screwdriver#2056

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by wahapo — APPROVED** [review]

**Review by yoshwata — APPROVED** [review]

**Review by ibu1224 — APPROVED** [review]

File: lib/pipeline.js [hunk 1]
```diff
@@ -639,6 +640,11 @@ class PipelineModel extends BaseModel {
             })
             .then(p => {
                 logger.info(`pipelineId:${this.id}: Creating child pipeline for ${scmUrl} with pipelineId:${p.id}.`);
+                // sync pipeline to create jobs
+                p.sync();
```
Comment (tkyi) [inline lib/pipeline.js]: does it need `await`?

File: lib/pipeline.js [hunk 2]
```diff
@@ -639,6 +640,11 @@ class PipelineModel extends BaseModel {
...
+                if (SD_API_URI) {
+                    p.addWebhook(`${SD_API_URI}/v4/webhooks`);
```
Comment (tkyi) [inline lib/pipeline.js]: same here
does it also need `await`?

Comment (jithine) [reply lib/pipeline.js]: were we not adding webhooks before ? I wonder how it's working in our setup 🤔

File: lib/pipeline.js (hunk 1 above)
Comment (kumada626) [reply lib/pipeline.js]: Yes, we should add `await`. I fixed it.

File: lib/pipeline.js (hunk 2 above)
Comment (kumada626) [reply lib/pipeline.js]: Currently, child pipelines are only created or updated when the parent pipeline is synced.
https://github.com/screwdriver-cd/models/blob/56e3ce9b02c68b99ca5848f3e9e029702f742f11/lib/pipeline.js#L730
However, child pipeline's webhooks are not added in this method.
The parent pipeline's webhook is added after it is created, but the child pipeline's ones are not.
https://github.com/screwdriver-cd/screwdriver/blob/cd5c538ba159f0229b7307f491eb4000892a47f4/plugins/pipelines/create.js#L120
So we should add webhooks after child pipeline is created.

**Review by tkyi — APPROVED** [review]

**Pull request merged**